File: BookAPIFrontend/src/app/models/book.ts
import { Injectable } from "@angular/core";

@Injectable({
  providedIn: 'root'
})
export class Book {

  Id: number = 0;
  Title: string = '';
  ISBN: string = '';
  Author: string = '';
  Price: number = 0;

  toStr() {
    return "Id = " + this.Id + ", Title = " + this.Title + ", ISBN = " + this.ISBN + ", Author = " + this.Author + ", Price = " + this.Price
  }
}

File: BookAPI/Models/Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace BookAPI.Models
{
    public class Book
    {
        public int Id { get; private set; }
        public string ISBN { get; set; }
        public string Title { get; set; }
        public String Author { get; set; }
        public float Price { get; set; }

        public Book()
        {
            ISBN = "";
            Title = "";
            Author = "";
            Price = 0;
        }

        public void SetId(int id)
        {
            this.Id = id;
        }

        public void PatchUp(Book book)
        {
            this.ISBN = (book.ISBN != null) ? book.ISBN : this.ISBN;
            this.Title = (book.Title != null) ? book.Title : this.Title;
            this.Author = (book.Author != null) ? book.Author : this.Author;
            this.Price = (book.Price != 0.0) ? book.Price : this.Price;
        }
    }
}